### DISCUSSION TITLE
Automatic sizing when nominal axis labels have angle 0?

### DISCUSSION THREAD

**Discussion by the21st** [vega/vega-lite#8412] [Q&A] [answered]
I built a simple bar chart with a nominal X axis type but I want to have the labels (country names) non-rotated for good readability. When I specify that, there's overlap between the labels. Is there any way for vega lite to automatically make the chart wider to accommodate for the axis labels?

[Vega editor spec](https://vega.github.io/editor/#/url/vega-lite/N4IgJAzgxgFgpgWwIYgFwhgF0wBwqgegIDc4BzJAOjIEtMYBXAI0poHsDp5kTykBaADZ04JAKyUAVhDYA7EABoQAEzjQATjRyZ289AEEABBBoIcguIaZJ1h2DcyGA7nRiHETOMtXLDypJhUiioBKKigxEiCDGpoANqgAMJsDLKY6gCeaCAAYupIslBwwfqk+WSWbABmhkgVaABMABwAvgpJKWmZ2QDicOrIsllKpf11lTXjaGJibR2p6VnoAMo4SDTyI2XjhtW19agALADMLQC6bSDI6gDW2dbqwXCFbMobZGigAB6fIFU0cEEymyyQW3SUmAyOGK6FkbAQGyiwSQXxoEF+giQnkE+lkZAsaAADC1LktQP9AcCDNsKrtJvUIVCYSAAI4MAo6QI6UggEktIA)

**Comment by jwoLondon** [comment, marked as answer]
Try setting `"width":"container"`. 

But I'd argue it is not often (if ever?) desirable to resize by label width. Do you really want your bar width to be determined by the label that takes up the most space in your categorical variable? I'd suggest that the negative visual impact of bars that are too 'fat' might outweigh the benefits of full text for every label.

Here are some other possible strategies:

* Setting `labelOverlap` to `true` will prevent the text from overplotting as in your example by labelling only those bars that don't result in overlapping text. For many nominal data this may not be desirable if the missing labels cannot be easily guessed.
* Reversing the x and y encoding so the long horizontal labels are parallel to the length of the bars.
* Explicitly setting the width of the chart to something that you can be confident would allow all labels to be shown.
* Setting `labelLimit` to some appropriate value (try 30 in your example) – this will truncate longer labels with an ellipsis, which may be sufficient for the label to be inferred. Can be combined with explicitly setting the width to some value larger than the default.
* Setting `labelFontSize` to be sufficiently small to fit labels (e.g. setting the chart width to 90 and label font size to 8 works reasonably well in your example). Depending on how you are creating your VL specs, these values could be calculated programmatically.
